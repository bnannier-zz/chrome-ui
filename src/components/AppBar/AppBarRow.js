import React, { Component } from "react";
import PropTypes from "prop-types";
import { Base } from "../";
import "./appbar.css";

/**
 * @description AppBarRow acts as the subcontainer of AppBar.
 * @param {string} id - The id attribute specifies a unique id for an HTML element (the value must be unique within the HTML document).
 * @param {string} className - CSS class that will be overriding this components style.
 * @param {node} children - Child component to be encapsulated by this component.
 */
class AppBarRow extends Component {
  render() {
    return (
      <Base {...this.props}>
        <div className={this.defaultCSS()}>{this.props.children}</div>
      </Base>
    );
  }

  defaultCSS() {
    return ["mdc-top-app-bar__row"];
  }
}

AppBarRow.propTypes = {
  id: PropTypes.string,
  className: PropTypes.string,
  children: PropTypes.node
};

export default AppBarRow;
