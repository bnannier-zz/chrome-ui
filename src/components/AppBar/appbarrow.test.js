import React from "react";
import { configure, shallow } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
configure({ adapter: new Adapter() });

import AppBarRow from "./AppBarRow";

describe("AppBarRow Tests", () => {
  it("Render AppBarRow", function() {
    const wrapper = shallow(
      <div>
        <AppBarRow>Chrome UI</AppBarRow>
      </div>
    );
    expect(wrapper).toMatchSnapshot();
  });
});
