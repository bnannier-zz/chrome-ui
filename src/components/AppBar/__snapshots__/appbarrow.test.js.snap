// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppBarRow Tests Render AppBarRow 1`] = `
<div>
  <AppBarRow>
    Chrome UI
  </AppBarRow>
</div>
`;
